frontend/src/features/admin-form/create/payment/PaymentDrawer.tsx

```diff
@@ -69,7 +69,13 @@ export const PaymentInput = (): JSX.Element => {
     handleSubmit,
   } = useForm<FormPayments>({
     mode: 'onBlur',
-    defaultValues: paymentsData,
+    defaultValues: {
+      ...paymentsData,
+      // Change payment_amount value to 2 decimal places temporarily for the form
+      payment_amount: paymentsData?.payment_amount
+        ? paymentsData?.payment_amount / 100
+        : 0,
+    },
```

Comment:
Rather than dealing with floating point numbers throughout the frontend, I wonder if it would be more appropriate to always keep an integer internal state, and then only convert the integer to it's currency representation at the point of rendering to the user. Not fully investigated, just the skeleton of the idea:

```typescript
// utils
const monetizeInteger = (val: number): string => ... // convert to "$xxx.xx" format

// input
const onChange = (val: string) => {
  // Strip out all non-digit characters
  const amount_cents = Number(val.replace(/\D/g,''))
  // Recompute correct value
  ... monetizeInteger(amount_cents)
}

Reply:
I think for form control, the value in `payments` object is directly tied to the value shown in the input box(?) That's why I did the conversion before loading and before saving the values. Lemme see what I can do to properly type it.

Reply:
Oh for this, I can't seem to find a way to do this easily... So I simply kept the typing used for the form values displayed (`FormPaymentsDisplay`) and form loading / submission (`FormPayments`) separate.